packages/regl-worldview/src/stories/assertionStories/Worldview.stories.js

```diff
@@ -0,0 +1,155 @@
+// @flow
+
+//  Copyright (c) 2018-present, GM Cruise LLC
+//
+//  This source code is licensed under the Apache License, Version 2.0,
+//  found in the LICENSE file in the root directory of this source tree.
+//  You may not use this file except in compliance with the License.
+
+/* eslint-disable react/display-name */
+
+import { storiesOf } from "@storybook/react";
+import React from "react";
+import { withScreenshot } from "storybook-chrome-screenshot";
+
+import Cubes from "../../commands/Cubes";
+import { WorldviewWrapper, clickAtOrigin, WORLDVIEW_SIZE } from "../worldviewAssertionUtils";
+import { assertionTest, expect, timeout } from "stories/assertionTestUtils";
+
+const cube = {
+  pose: {
+    orientation: { x: 0, y: 0, z: 0, w: 1 },
+    position: { x: 0, y: -20, z: 0 },
+  },
+  scale: { x: 10, y: 10, z: 10 },
+  color: { r: 1, g: 0, b: 1, a: 0.5 },
+};
+
+function createClickFunction(eventName): () => Promise<void> {
+  return async () => {
+    const [element] = document.getElementsByTagName("canvas");
+    if (!element) {
+      throw new Error("Could not find canvas element");
+    }
+    const mouseEvent = new MouseEvent(eventName, {
+      bubbles: true,
+      clientX: WORLDVIEW_SIZE / 2,
+      clientY: WORLDVIEW_SIZE / 2,
+    });
+    element.dispatchEvent(mouseEvent);
+    await timeout(100);
+  };
+}
+
+const mouseMove = createClickFunction("mousemove");
+const mouseUp = createClickFunction("mouseup");
+const mouseDown = createClickFunction("mousedown");
+const doubleClick = createClickFunction("dblclick");
```

Comment:
Maybe rename the function above `emitClickEvent(eventName: "mousemove" | "mouseup" | "mousedown" | "dblclick"): Promise<void>`, remove the function nesting, and then just call it as `emitClickEvent("mousemove");`? Then you don't need the indirection.